astropy/units/function/core.py

```diff
@@ -571,23 +571,35 @@ def __array_ufunc__(self, function, method, *inputs, **kwargs):
         return super().__array_ufunc__(function, method, *inputs, **kwargs)
 
     # ↓↓↓ methods overridden to change behavior
+    def _maybe_new_view(self, result):
+        """View as function quantity if the unit is unchanged.
+
+        Used for the case that self.unit.physical_unit is dimensionless,
+        where multiplication and division is done using the Quantity
+        equivalent, to transform them back to a FunctionQuantity if possible.
+        """
+        if isinstance(result, Quantity) and result.unit == self.unit:
```

Comment:
Wouldn't this be faster?
```suggestion
        if isinstance(result, Quantity) and result.unit is self.unit:
```

Reply:
the `is` only works for `NamedUnit` instances, not for composite ones.

Reply:
That said, in this case presumably the unit is set directly with the old one...

Reply:
Ah, no, the case we have here is comparing `u.mag()` with `u.mag`, which are not the same thing though they compare equal.